registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/eventbus/KafkaMosipEventBus.java

```diff
@@ -429,4 +432,60 @@ private void resumePartition(TopicPartition topicPartition, Promise<Void> promis
 				promise.fail("Partition resuming failed for " + topicPartition.getPartition());
 		});
 	}
+
+	@Override
+	public void consumerHealthCheck(EventHandler<EventDTO, Handler<AsyncResult<HealthCheckDTO>>> eventHandler,
+			String address) {
+		EventDTO eventDTO = new EventDTO();
+		HealthCheckDTO healthCheckDTO = new HealthCheckDTO();
+		kafkaConsumer.listTopics(f -> {
+			if (f.succeeded()) {
+				healthCheckDTO.setEventBusConnected(true);
+
+			} else {
+				healthCheckDTO.setEventBusConnected(false);
+				healthCheckDTO.setFailureReason(f.cause().getMessage());
+			}
+			try {
+				eventDTO.setBody(new JsonObject(JsonUtil.objectMapperObjectToJson(healthCheckDTO)));
+			} catch (IOException e) {
+				healthCheckDTO.setEventBusConnected(false);
+				healthCheckDTO.setFailureReason(e.getMessage());
+				eventHandler.handle(eventDTO, res -> {
+				});
+			}
+			eventHandler.handle(eventDTO, res -> {
+
+			});
+		});
+
+	}
+
+	@Override
+	public void sendHealthCheck(EventHandler<EventDTO, Handler<AsyncResult<HealthCheckDTO>>> eventHandler,

```

Comment:
Lets rename this method to producerHealthCheck or senderHealthCheck

Reply:
fixed